src/layouting/data_limits.jl

```diff
@@ -177,103 +162,15 @@ function update_boundingbox!(bb_ref, bb::Rect)
-end
-
-function _update_rect(rect::Rect{N, T}, point::VecTypes{N, T}) where {N, T}
-    mi = minimum(rect)
-    ma = maximum(rect)
-    mis_mas = map(mi, ma, point) do _mi, _ma, _p
-        (isnan(_mi) ? _p : _p < _mi ? _p : _mi), (isnan(_ma) ? _p : _p > _ma ? _p : _ma)
-    end
-    new_o = map(first, mis_mas)
-    new_w = map(mis_mas) do (mi, ma)
-        ma - mi
-    end
-    typeof(rect)(new_o, new_w)
-end
-
-function limits_from_transformed_points(points_iterator)
-    isempty(points_iterator) && return Rect3d()
-    first, rest = Iterators.peel(points_iterator)
-    bb = foldl(_update_rect, rest, init = Rect3d(first, zero(first)))
-    return bb
-end
-
-# include bbox from scaled markers
-function limits_from_transformed_points(positions, scales, rotations, element_bbox)
-    isempty(positions) && return Rect3d()
-
-    first_scale = attr_broadcast_getindex(scales, 1)
-    first_rot = attr_broadcast_getindex(rotations, 1)
-    full_bbox = Ref(first_rot * (element_bbox * first_scale) + first(positions))
-    for (i, pos) in enumerate(positions)
-        scale, rot = attr_broadcast_getindex(scales, i), attr_broadcast_getindex(rotations, i)
-        transformed_bbox = rot * (element_bbox * scale) + pos
-        update_boundingbox!(full_bbox, transformed_bbox)
-    end
-
-    return full_bbox[]
-end
-
-function data_limits(scenelike, exclude=(p)-> false)
-    bb_ref = Base.RefValue(Rect3d())
-    foreach_plot(scenelike) do plot
-        if !exclude(plot)
-            update_boundingbox!(bb_ref, data_limits(plot))
-        end
-    end
-    return bb_ref[]
-end
-
-# A few overloads for performance
-function data_limits(plot::Surface)
-    mini_maxi = extrema_nan.((plot.x[], plot.y[], plot.z[]))
-    mini = first.(mini_maxi)
-    maxi = last.(mini_maxi)
-    return Rect3d(mini, maxi .- mini)
-end
-
-function data_limits(plot::Heatmap)
-    mini_maxi = extrema_nan.((plot.x[], plot.y[]))
-    mini = Vec3d(first.(mini_maxi)..., 0)
-    maxi = Vec3d(last.(mini_maxi)..., 0)
-    return Rect3d(mini, maxi .- mini)
-end
-
-function data_limits(plot::Image)
-    mini_maxi = extrema_nan.((plot.x[], plot.y[]))
-    mini = Vec3d(first.(mini_maxi)..., 0)
-    maxi = Vec3d(last.(mini_maxi)..., 0)
-    return Rect3d(mini, maxi .- mini)
-end
-
-function data_limits(plot::MeshScatter)
-    # TODO: avoid mesh generation here if possible
-    @get_attribute plot (marker, markersize, rotations)
-    marker_bb = Rect3d(marker)
-    positions = iterate_transformed(plot)
-    scales = markersize
-    # fast path for constant markersize
-    if scales isa VecTypes{3} && rotations isa Quaternion
-        bb = limits_from_transformed_points(positions)
-        marker_bb = rotations * (marker_bb * scales)
-        return Rect3d(minimum(bb) + minimum(marker_bb), widths(bb) + widths(marker_bb))
+foreach_plot(f, s::Scene) = foreach_plot(f, s.plots)
+# foreach_plot(f, s::Figure) = foreach_plot(f, s.scene)
+# foreach_plot(f, s::FigureAxisPlot) = foreach_plot(f, s.figure)
```

Comment:
It doesn't make sense for data_limits or boundingbox to consider child scenes, as they can have different transforms and cameras. These changes are meant exclude them. Searching the repo doesn't show any other uses for these so I hope this is fine